### DISCUSSION TITLE
Recognize language

### DISCUSSION THREAD

**Discussion by jorge2985** [community/community#23847] [New to GitHub]
<p>Hi, sorry but I didn’t find the next question. How can to recognize Github a new programming language? We are a foundation who developed and would like to appear in Github. I don’t know if I was clear.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>GitHub uses Linguist to detect languages, see the “Adding a language” section in the Linguist contributing guidelines:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/github/linguist/blob/971e152fabca12e0ad386833eecb7b15e312a09f/CONTRIBUTING.md#adding-a-language">
  <header class="source">

      <a href="https://github.com/github/linguist/blob/971e152fabca12e0ad386833eecb7b15e312a09f/CONTRIBUTING.md#adding-a-language" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/github/linguist/blob/971e152fabca12e0ad386833eecb7b15e312a09f/CONTRIBUTING.md#adding-a-language" target="_blank" rel="noopener">github/linguist/blob/971e152fabca12e0ad386833eecb7b15e312a09f/CONTRIBUTING.md#adding-a-language</a></h4>


    <pre><code class="lang-md"># Contributing

Hi there!
We're thrilled that you'd like to contribute to this project.
Your help is essential for keeping it great.

Contributions to this project are [released](https://docs.github.com/github/site-policy/github-terms-of-service#6-contributions-under-repository-license) to the public under the [project's open source license](LICENSE).

This project adheres to the [Contributor Covenant Code of Conduct](http://contributor-covenant.org/).
By participating, you are expected to uphold this code.

The majority of contributions won't need to touch any Ruby code at all.

## Dependencies

Linguist is a Ruby library so you will need a recent version of Ruby installed.
There are known problems with the macOS/XCode supplied version of Ruby that causes problems installing some of the dependencies.
Accordingly, we highly recommend you install a version of Ruby using Homebrew, `rbenv`, `rvm`, `ruby-build`, `asdf` or other packaging system, before attempting to install Linguist and the dependencies.

Linguist uses the [`charlock_holmes`](https://github.com/brianmario/charlock_holmes) character encoding detection library which in turn uses [ICU](http://site.icu-project.org/), and the libgit2 bindings for Ruby provided by [`rugged`](https://github.com/libgit2/rugged).
</code></pre>


  This file has been truncated. <a href="https://github.com/github/linguist/blob/971e152fabca12e0ad386833eecb7b15e312a09f/CONTRIBUTING.md#adding-a-language" target="_blank" rel="noopener">show original</a>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
